schema.graphql

```diff
@@ -1,122 +1,124 @@
 type Vault @entity {
   id: ID!
-  "Amount of underlying token per 1 share"
-  pricePerFullShare: BigDecimal!
-  "Total shares supply"
-  totalSupply: BigDecimal!
-  "Full vault underlying token balance (vault + strategy)"
-  vaultBalance: BigDecimal!
-  "Strategy underlying token balance"
-  strategyBalance: BigDecimal!
-  # TODO:
-  "How much the vault allows to be borrowed"
-  available: BigDecimal!
-  "Deposit token"
+  "Deposit token info"
   underlyingToken: Token!
+  "Share token (ytoken) info"
   shareToken: Token!
-  currentController: Controller!
-  currentStrategy: Strategy!
-  "Transaction metadata for the last update"
+  "Current controller"
+  controller: Controller!
+  "Current strategy"
+  strategy: Strategy!
+  # TODO: replace with tx when was created
+  "Creation transaction"
   transaction: Transaction!
-  # TODO:
-  "balance: totalDeposited - totalWithdrawn: all deposits of underlying made by external accounts"
-  netDeposits: BigDecimal!
-  totalDeposited: BigDecimal!
-  totalWithdrawn: BigDecimal!
-  "totalActiveShares: totalSharesMinted - totalSharesBurned"
-  totalActiveShares: BigDecimal!
-  totalSharesMinted: BigDecimal!
-  totalSharesBurned: BigDecimal!
-  totalEarnings: BigDecimal!
-  # raw
-  pricePerFullShareRaw: BigInt!
-  totalSupplyRaw: BigInt!
-  # TODO:
-  "Balance of the Vault contract of underlying Token + balance of the Strategy contract of underlying Token"
-  vaultBalanceRaw: BigInt!
-  # TODO:
-  "Balance of underlying Token specifically held in the strategy"
-  strategyBalanceRaw: BigInt!
-  # TODO:
-  "How much the vault allows to be borrowed"
-  availableRaw: BigInt!
-  netDepositsRaw: BigInt!
-  totalDepositedRaw: BigInt!
-  totalWithdrawnRaw: BigInt!
-  totalActiveSharesRaw: BigInt!
-  totalSharesMintedRaw: BigInt!
-  totalSharesBurnedRaw: BigInt!
-  totalEarningsRaw: BigInt!
-  totalHarvestCalls: BigInt!
+
+  vaultBalanceUpdates: [VaultBalanceUpdate!]! @derivedFrom(field: "vault")
+
+  vaultUpdates: [VaultUpdate!]!
+  strategyUpdates: [StrategyUpdate!]!
+
   # derived fields
   transfers: [Transfer!]! @derivedFrom(field: "vault")
   deposits: [Deposit!]! @derivedFrom(field: "vault")
   withdrawals: [Withdrawal!]! @derivedFrom(field: "vault")
   harvests: [Harvest!]! @derivedFrom(field: "vault")
   balances: [AccountVaultBalance!]! @derivedFrom(field: "vault")
   strategies: [Strategy!]! @derivedFrom(field: "vault")
-  controllers: [Controller!]! @derivedFrom(field: "vault")
+  # controllers: [Controller!]! @derivedFrom(field: "vault")
+}
+
+# used to track balance of both user and vault
+interface BalanceUpdate @entity {
+  id: ID!
+  transaction: Transaction!
+
+  netDepositsRaw: BigInt!
+  totalDepositRaw: BigInt!
+  totalWithdrawalRaw: BigInt!
+  totalSharesMintedRaw: BigInt!
+  totalSharesBurnedRaw: BigInt!
+}
+
+type VaultBalanceUpdate implements BalanceUpdate @entity {
+  id: ID!
+  transaction: Transaction!
+
+  netDepositsRaw: BigInt!
```

Comment:
Why do we utilize "Raw?" What is non-raw? Could potentially remove Raw suffix everywhere. It is confusing